src/CommunityToolkit.Maui.Camera/CameraManager.macios.cs

```diff
@@ -191,42 +196,51 @@ protected virtual async partial ValueTask PlatformTakePicture(CancellationToken
 
 		var capturePhotoSettings = AVCapturePhotoSettings.FromFormat(codecSettings);
 		capturePhotoSettings.FlashMode = photoOutput.SupportedFlashModes.Contains(flashMode) ? flashMode : photoOutput.SupportedFlashModes.First();
+		var avMediaTypeVideo = AVMediaTypes.Video.GetConstant();
+		if (avMediaTypeVideo is not null)
+		{
+			var photoOutputConnection = photoOutput.ConnectionFromMediaType(avMediaTypeVideo);
+			if (photoOutputConnection is not null)
+			{
+				photoOutputConnection.VideoOrientation = videoOrientation;
+			}
+		}
 
 		var wrapper = new AVCapturePhotoCaptureDelegateWrapper();
 
 		photoOutput.CapturePhoto(capturePhotoSettings, wrapper);
 
 		var result = await wrapper.Task.WaitAsync(token);
-		var data = result.Photo.FileDataRepresentation;
-
 		if (result.Error is not null)
 		{
-			cameraView.OnMediaCapturedFailed(result.Error.LocalizedFailureReason);
-			return;
-		}
+			var failureReason = result.Error.LocalizedDescription;
+			if (!string.IsNullOrEmpty(result.Error.LocalizedFailureReason))
+			{
+				failureReason = $"{failureReason} - {result.Error.LocalizedFailureReason}";
+			}
 
-		if (data is null)
-		{
-			cameraView.OnMediaCapturedFailed("Unable to retrieve the file data representation from the captured result.");
+			cameraView.OnMediaCapturedFailed(failureReason);
 			return;
 		}
 
-		var dataBytes = ArrayPool<byte>.Shared.Rent((int)data.Length);
-
+		Stream? imageData;
 		try
 		{
-			Marshal.Copy(data.Bytes, dataBytes, 0, (int)data.Length);
-
-			cameraView.OnMediaCaptured(new MemoryStream(dataBytes));
+			imageData = result.Photo.FileDataRepresentation?.AsStream();
 		}
-		catch (Exception ex)
+		catch
```

Comment:
Would there be any value in including the exception details in the `OnMediaCaptureFailed` call?